addons/casings/functions/fnc_createCasing.sqf

```diff
@@ -0,0 +1,69 @@
+#include "script_component.hpp"
+/*
+ * Author: esteldunedain / Cyruz / Diwako
+ * Produces a casing matching the fired rifles caliber on the ground around the unit
+ *
+ * Arguments:
+ * 0: unit - Object the event handler is assigned to <OBJECT>
+ * 1: weapon - Fired weapon <STRING>
+ * 2: ammo - Ammo used <STRING>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [player, currentWeapon player, "","", player ammo currentWeapon player] call ace_casings_fnc_createCase
+ *
+ * Public: No
+ */
+
+params ["_unit", "_weapon", "", "", "_ammo"];
+
+if (!isNull objectParent _unit || { _weapon in ["Throw","Put"] } ) exitWith {};
+
+private _modelPath = GVAR(casings) get _ammo;
+
+if (isNil "_modelPath") then {
+    private _cartridge = getText (configFile >> "CfgAmmo" >> _ammo >> "cartridge");
+    //Default cartridge is a 5.56mm model
+    _modelPath = switch (_cartridge) do {
+        case "FxCartridge_9mm":     { "A3\Weapons_f\ammo\cartridge_small.p3d" };
+        case "FxCartridge_65":      { "A3\weapons_f\ammo\cartridge_65.p3d" };
+        case "FxCartridge_762":     { "A3\weapons_f\ammo\cartridge_762.p3d" };
+        case "FxCartridge_127":     { "A3\weapons_f\ammo\cartridge_127.p3d" };
+        case "FxCartridge_slug":    { "A3\weapons_f\ammo\cartridge_slug.p3d" };
+        case "":                    { "" };
+        default { "A3\Weapons_f\ammo\cartridge.p3d" };
```

Comment:
Could probably be optimized using HashMap.

Also, getText and building the config is kind of slow. Since this is basically every frame for fast firing weapons: caching?

Reply:
Ah, nvm, it already is cached. Probably want to rename `GVAR(casings)` to something that says it's an cache (and replace GVAR(casingsArr) with just GVAR(casings)).

Reply:
Switched to cachedCasings for hashmap and casings for the array.